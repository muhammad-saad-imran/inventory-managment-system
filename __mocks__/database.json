{
  "clients": [
    {
      "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
      "name": "Alice Smith",
      "email": "alice.smith@example.com",
      "phone_number": 1234567890,
      "created_at": "2024-01-01T12:00:00Z"
    },
    {
      "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
      "name": "Bob Johnson",
      "email": "bob.johnson@example.com",
      "phone_number": 9876543210,
      "created_at": "2024-01-05T15:30:00Z"
    }
  ],
  "products": [
    {
      "id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
      "name": "Product A",
      "description": "Description of Product A",
      "created_at": "2024-02-01T09:00:00Z"
    },
    {
      "id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
      "name": "Product B",
      "description": "Description of Product B",
      "created_at": "2024-02-15T10:00:00Z"
    }
  ],
  "suppliers": [
    {
      "id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
      "name": "Supplier One",
      "email": "supplier.one@example.com",
      "phone_number": 5551234567,
      "created_at": "2024-03-01T08:00:00Z"
    },
    {
      "id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
      "name": "Supplier Two",
      "email": "supplier.two@example.com",
      "phone_number": 5559876543,
      "created_at": "2024-03-10T14:00:00Z"
    }
  ],
  "inventory": [
    {
      "id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
      "cost": 50.00,
      "supplier_id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
      "product_id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
      "stock_quantity": 100,
      "reorder_limit": 20,
      "supply_date": "2024-03-15T10:00:00Z",
      "created_at": "2024-03-15T10:00:00Z",
      "suppliers": {
        "id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
        "name": "Supplier One",
        "email": "supplier.one@example.com",
        "phone_number": 5551234567,
        "created_at": "2024-03-01T08:00:00Z"
      },
      "products": {
        "id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
        "name": "Product A",
        "description": "Description of Product A",
        "created_at": "2024-02-01T09:00:00Z"
      }
    },
    {
      "id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
      "cost": 75.00,
      "supplier_id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
      "product_id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
      "stock_quantity": 50,
      "reorder_limit": 10,
      "supply_date": "2024-03-20T11:00:00Z",
      "created_at": "2024-03-20T11:00:00Z",
      "suppliers": {
        "id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
        "name": "Supplier Two",
        "email": "supplier.two@example.com",
        "phone_number": 5559876543,
        "created_at": "2024-03-10T14:00:00Z"
      },
      "products": {
        "id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
        "name": "Product B",
        "description": "Description of Product B",
        "created_at": "2024-02-15T10:00:00Z"
      }
    }
  ],
  "orders": [
    {
      "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
      "client_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
      "order_date": "2024-04-01T12:00:00Z",
      "status": "PROCESSING",
      "created_at": "2024-04-01T12:00:00Z",
      "clients": {
        "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
        "name": "Alice Smith",
        "email": "alice.smith@example.com",
        "phone_number": 1234567890,
        "created_at": "2024-01-01T12:00:00Z"
      },
      "order_items": [
        {
          "id": "d8e10f0c-bb27-4d8d-bff6-46f9e925a5b1",
          "order_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
          "inventory_id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
          "quantity": 2,
          "price": 100.00,
          "created_at": "2024-04-01T12:00:00Z",
          "inventory": {
            "id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
            "cost": 50.00,
            "supplier_id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
            "product_id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
            "stock_quantity": 100,
            "reorder_limit": 20,
            "supply_date": "2024-03-15T10:00:00Z",
            "created_at": "2024-03-15T10:00:00Z",
            "suppliers": {
              "id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
              "name": "Supplier One",
              "email": "supplier.one@example.com",
              "phone_number": 5551234567,
              "created_at": "2024-03-01T08:00:00Z"
            },
            "products": {
              "id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
              "name": "Product A",
              "description": "Description of Product A",
              "created_at": "2024-02-01T09:00:00Z"
            }
          }
        }
      ]
    },
    {
      "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
      "client_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
      "order_date": "2024-04-05T15:30:00Z",
      "status": "SHIPPED",
      "created_at": "2024-04-05T15:30:00Z",
      "clients": {
        "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
        "name": "Bob Johnson",
        "email": "bob.johnson@example.com",
        "phone_number": 9876543210,
        "created_at": "2024-01-05T15:30:00Z"
      },
      "order_items": [
        {
          "id": "d8e10f0c-bb27-4d8d-bff6-46f9e925a5b2",
          "order_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
          "inventory_id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
          "quantity": 1,
          "price": 75.00,
          "created_at": "2024-04-05T15:30:00Z",
          "inventory": {
            "id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
            "cost": 75.00,
            "supplier_id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
            "product_id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
            "stock_quantity": 50,
            "reorder_limit": 10,
            "supply_date": "2024-03-20T11:00:00Z",
            "created_at": "2024-03-20T11:00:00Z",
            "suppliers": {
              "id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
              "name": "Supplier Two",
              "email": "supplier.two@example.com",
              "phone_number": 5559876543,
              "created_at": "2024-03-10T14:00:00Z"
            },
            "products": {
              "id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
              "name": "Product B",
              "description": "Description of Product B",
              "created_at": "2024-02-15T10:00:00Z"
            }
          }
        }
      ]
    }
  ],
  "invoices": [
    {
      "id": "c7f9d0d5-9a30-4a26-b0c4-00ff18b8b781",
      "order_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
      "amount": 200.00,
      "invoice_date": "2024-04-02T12:00:00Z",
      "order": {
        "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
        "client_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
        "order_date": "2024-04-01T12:00:00Z",
        "status": "PROCESSING",
        "created_at": "2024-04-01T12:00:00Z",
        "clients": {
          "id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7510",
          "name": "Alice Smith",
          "email": "alice.smith@example.com",
          "phone_number": 1234567890,
          "created_at": "2024-01-01T12:00:00Z"
        },
        "order_items": [
          {
            "id": "d8e10f0c-bb27-4d8d-bff6-46f9e925a5b1",
            "order_id": "e4b6a0f1-d09e-4e95-8d1a-8e091b2a7511",
            "inventory_id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
            "quantity": 2,
            "price": 100.00,
            "created_at": "2024-04-01T12:00:00Z",
            "inventory": {
              "id": "edc1b083-5c7f-4d7f-8d52-1c5302a3d7b1",
              "cost": 50.00,
              "supplier_id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
              "product_id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
              "stock_quantity": 100,
              "reorder_limit": 20,
              "supply_date": "2024-03-15T10:00:00Z",
              "created_at": "2024-03-15T10:00:00Z",
              "suppliers": {
                "id": "b9d6d220-4f6c-42d8-9083-b0a5c3f05a9a",
                "name": "Supplier One",
                "email": "supplier.one@example.com",
                "phone_number": 5551234567,
                "created_at": "2024-03-01T08:00:00Z"
              },
              "products": {
                "id": "f2b2d5eb-76d4-4b09-a64d-9c2fd6a4d19e",
                "name": "Product A",
                "description": "Description of Product A",
                "created_at": "2024-02-01T09:00:00Z"
              }
            }
          }
        ]
      }
    },
    {
      "id": "a9f5d0d4-90ff-4a69-b0b5-7b8e3c5fbb9d",
      "order_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
      "amount": 75.00,
      "invoice_date": "2024-04-06T15:30:00Z",
      "order": {
        "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
        "client_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
        "order_date": "2024-04-05T15:30:00Z",
        "status": "SHIPPED",
        "created_at": "2024-04-05T15:30:00Z",
        "clients": {
          "id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d1",
          "name": "Bob Johnson",
          "email": "bob.johnson@example.com",
          "phone_number": 9876543210,
          "created_at": "2024-01-05T15:30:00Z"
        },
        "order_items": [
          {
            "id": "d8e10f0c-bb27-4d8d-bff6-46f9e925a5b2",
            "order_id": "c3f7d832-04e1-4b77-b19b-5a7c36be83d2",
            "inventory_id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
            "quantity": 1,
            "price": 75.00,
            "created_at": "2024-04-05T15:30:00Z",
            "inventory": {
              "id": "ac15d5a6-14f1-4a7f-98a3-0e4f2e1c1d4f",
              "cost": 75.00,
              "supplier_id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
              "product_id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
              "stock_quantity": 50,
              "reorder_limit": 10,
              "supply_date": "2024-03-20T11:00:00Z",
              "created_at": "2024-03-20T11:00:00Z",
              "suppliers": {
                "id": "a2b5d5b9-4c3a-4b51-a2c6-123456789abc",
                "name": "Supplier Two",
                "email": "supplier.two@example.com",
                "phone_number": 5559876543,
                "created_at": "2024-03-10T14:00:00Z"
              },
              "products": {
                "id": "a4cde3f2-1f98-4b53-b8ee-8f7d64a84b9d",
                "name": "Product B",
                "description": "Description of Product B",
                "created_at": "2024-02-15T10:00:00Z"
              }
            }
          }
        ]
      }
    }
  ]
}
