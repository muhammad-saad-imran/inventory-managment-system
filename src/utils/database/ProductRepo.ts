import { GenericRepo } from "@/utils/database/GenericRepo";
import { DB_TABLES, Product } from "@/utils/database/types";
import { SupabaseClient } from "@supabase/supabase-js";

export class ProductRepo extends GenericRepo<Product> {
  constructor(_client: SupabaseClient) {
    super(_client, DB_TABLES.PRODUCTS);
  }

  async getWithName(name: string, include?: string): Promise<Product[]> {
    const { data: entity, error } = await this.supabase
      .from(this.tableName)
      .select(include)
      .ilike("name", `%${name}%`);

    if (entity) {
      return entity as unknown as Product[];
    } else {
      throw error;
    }
  }
}
