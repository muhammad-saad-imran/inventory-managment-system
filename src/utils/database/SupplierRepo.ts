import { GenericRepo } from "@/utils/database/GenericRepo";
import { DB_TABLES, Supplier } from "@/utils/database/types";
import { SupabaseClient } from "@supabase/supabase-js";

export class SupplierRepo extends GenericRepo<Supplier> {
  constructor(_client: SupabaseClient) {
    super(_client, DB_TABLES.SUPPLIERS);
  }

  async getWithName(name: string, include?: string): Promise<Supplier[]> {
    const { data: entity, error } = await this.supabase
      .from(this.tableName)
      .select(include)
      .ilike("name", `%${name}%`);

    if (entity) {
      return entity as unknown as Supplier[];
    } else {
      throw error;
    }
  }
}
