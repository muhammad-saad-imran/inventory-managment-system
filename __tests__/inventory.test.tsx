import { act, cleanup } from "@testing-library/react";
import { fireEvent, screen } from "@testing-library/dom";
import selectEvent from "react-select-event";
import { renderWithProviders } from "@/utils/test-utils";
import { InventoryRepo } from "@/utils/database/InventoryRepo";
import { ProductRepo } from "@/utils/database/ProductRepo";
import { SupplierRepo } from "@/utils/database/SupplierRepo";
import database from "@/../__mocks__/database.json";
import InventoryPage from "@/app/dashboard/inventory/page";
import CreateInventoryPage from "@/app/dashboard/inventory/create/page";
import InventoryInfoPage from "@/app/dashboard/inventory/[id]/page";

jest.mock("next/navigation", () => {
  const actualImport = jest.requireActual("next/navigation");
  return {
    ...actualImport,
    usePathname() {
      return "/dashboard/suppliers";
    },
  };
});

jest.mock("@/utils/supabase/client");
jest.mock("@/utils/database/InventoryRepo");
jest.mock("@/utils/database/ProductRepo");
jest.mock("@/utils/database/SupplierRepo");

describe("Inventory features", () => {
  // Inventory Repo spies
  let getWithProductNameMock = jest.spyOn(
    InventoryRepo.prototype,
    "getWithProductName"
  );
  let getInventoryMock = jest.spyOn(InventoryRepo.prototype, "get");
  let createInventoryMock = jest.spyOn(InventoryRepo.prototype, "create");
  let updateInventoryMock = jest.spyOn(InventoryRepo.prototype, "update");
  let deleteInventoryMock = jest.spyOn(InventoryRepo.prototype, "delete");

  // Product Repo spies
  let getProductWithNameMock = jest.spyOn(ProductRepo.prototype, "getWithName");

  // Supplier repo spies
  let getSupplierWithNameMock = jest.spyOn(
    SupplierRepo.prototype,
    "getWithName"
  );

  let useRouter = jest.spyOn(require("next/navigation"), "useRouter");
  let useParams = jest.spyOn(require("next/navigation"), "useParams");
  const routerPushMock = jest.fn();

  beforeEach(() => {
    // mock useRouter implementation
    useRouter.mockImplementation(() => ({
      push: routerPushMock,
    }));

    // mock useParams implementation
    useParams.mockImplementation(() => ({ id: database.inventory[0].id }));

    // mock Inventory Repo methods
    getWithProductNameMock.mockResolvedValue(database.inventory);
    getInventoryMock.mockResolvedValue(database.inventory[0]);
    createInventoryMock.mockResolvedValue(database.inventory[0]);
    updateInventoryMock.mockResolvedValue(database.inventory[0]);

    // mock Product repo methods
    getProductWithNameMock.mockResolvedValue(database.products);

    // mock Supplier repo methods
    getSupplierWithNameMock.mockResolvedValue(database.suppliers);
  });

  afterEach(() => {
    // Cleanup mocks after each test
    getWithProductNameMock.mockClear();
    getInventoryMock.mockClear();
    createInventoryMock.mockClear();
    updateInventoryMock.mockClear();
    getProductWithNameMock.mockClear();
    getSupplierWithNameMock.mockClear();
    useRouter.mockClear();
    useParams.mockClear();

    cleanup();
  });

  describe("InventoryPage", () => {
    it("should match snapshot", async () => {
      let container;

      await act(async () => {
        ({ container } = await renderWithProviders(<InventoryPage />)); // ARRANGE
      });
      expect(container).toMatchSnapshot();
    });

    it("should route on clicking inventory table row", async () => {
      renderWithProviders(<InventoryPage />); // ARRANGE

      // ACT
      const elems = await screen.findAllByTestId("inventory-row");
      fireEvent.click(elems[0]);

      expect(routerPushMock).toHaveBeenCalledTimes(1); // ASSERT
    });
  });

  describe("CreateInventoryPage", () => {
    it("should match snapshot", () => {
      const { container } = renderWithProviders(<CreateInventoryPage />);
      expect(container).toMatchSnapshot();
    });
  });

  describe("InventoryInfoPage", () => {
    it("should match snapshot", () => {
      const { container } = renderWithProviders(<InventoryInfoPage />);
      expect(container).toMatchSnapshot();
    });

    it("should delete inventory", async () => {
      await act(() => renderWithProviders(<InventoryInfoPage />));

      await act(async () => {
        fireEvent.click(await screen.findByRole("button", { name: /delete/i }));
      });

      expect(deleteInventoryMock).toHaveBeenCalledTimes(1);
      expect(routerPushMock).toHaveBeenCalledTimes(1);
    });

    it("should update inventory", async () => {
      await act(() => renderWithProviders(<InventoryInfoPage />));

      await act(async () => {
        fireEvent.change(await screen.findByLabelText(/stock/i), {
          target: { value: 200 },
        });

        fireEvent.click(await screen.findByRole("button", { name: /update/i }));
      });

      expect(updateInventoryMock).toHaveBeenCalledTimes(1);
    });
  });
});
