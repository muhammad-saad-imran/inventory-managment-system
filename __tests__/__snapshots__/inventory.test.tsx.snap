// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory features CreateInventoryPage should match snapshot 1`] = `
<div>
  <form>
    <p
      class="text-3xl text-center"
    >
      Create Inventory
    </p>
    <div
      class="flex flex-col gap-3 w-full bg-white mt-3 p-3"
    >
      <div
        class="flex gap-4"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="Product"
          >
            Product
          </label>
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class=" css-13cymwt-control"
            >
              <div
                class=" css-hlgwow"
              >
                <div
                  class=" css-1dimb5e-singleValue"
                />
                <div
                  class=" css-19bb58m"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="Product"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1wy0on6"
              >
                <div
                  aria-hidden="true"
                  class=" css-pogzpp-loadingIndicator"
                >
                  <span
                    class="css-1h1juvu-LoadingDot"
                  />
                  <span
                    class="css-1swgd6a-LoadingDot"
                  />
                  <span
                    class="css-41hts2-LoadingDot"
                  />
                </div>
                <span
                  class=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-1xc3v61-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="Product"
              type="hidden"
              value=""
            />
          </div>
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="Supplier"
          >
            Supplier
          </label>
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-3-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class=" css-13cymwt-control"
            >
              <div
                class=" css-hlgwow"
              >
                <div
                  class=" css-1dimb5e-singleValue"
                />
                <div
                  class=" css-19bb58m"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="Supplier"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1wy0on6"
              >
                <div
                  aria-hidden="true"
                  class=" css-pogzpp-loadingIndicator"
                >
                  <span
                    class="css-1h1juvu-LoadingDot"
                  />
                  <span
                    class="css-1swgd6a-LoadingDot"
                  />
                  <span
                    class="css-41hts2-LoadingDot"
                  />
                </div>
                <span
                  class=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-1xc3v61-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="Supplier"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex gap-4"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="cost"
          >
            Cost
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="cost"
            id="cost"
            name="cost"
            type="number"
            value=""
          />
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="supply_date"
          >
            Supply date
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="supply_date"
            id="supply_date"
            name="supply_date"
            type="date"
            value=""
          />
        </div>
      </div>
      <div
        class="flex gap-4"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="stock_quantity"
          >
            Stock
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="stock_quantity"
            id="stock_quantity"
            name="stock_quantity"
            type="number"
            value="0"
          />
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="reorder_limit"
          >
            Reorder limit
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="reorder_limit"
            id="reorder_limit"
            name="reorder_limit"
            type="number"
            value="0"
          />
        </div>
      </div>
      <button
        class="sc-7d679c72-0 sc-7d679c72-1 zDViG iTQhfb w-fit"
        type="submit"
      >
        Create
      </button>
    </div>
  </form>
</div>
`;

exports[`Inventory features InventoryInfoPage should match snapshot 1`] = `
<div>
  <form>
    <p
      class="text-3xl text-center"
    >
      Create Inventory
    </p>
    <div
      class="flex flex-col gap-3 w-full bg-white mt-3 p-3"
    >
      <div
        class="flex gap-4"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="Product"
          >
            Product
          </label>
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-4-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class=" css-13cymwt-control"
            >
              <div
                class=" css-hlgwow"
              >
                <div
                  class=" css-1dimb5e-singleValue"
                />
                <div
                  class=" css-19bb58m"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="Product"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1wy0on6"
              >
                <div
                  aria-hidden="true"
                  class=" css-pogzpp-loadingIndicator"
                >
                  <span
                    class="css-1h1juvu-LoadingDot"
                  />
                  <span
                    class="css-1swgd6a-LoadingDot"
                  />
                  <span
                    class="css-41hts2-LoadingDot"
                  />
                </div>
                <span
                  class=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-1xc3v61-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="Product"
              type="hidden"
              value=""
            />
          </div>
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="Supplier"
          >
            Supplier
          </label>
          <div
            class=" css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-5-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class=" css-13cymwt-control"
            >
              <div
                class=" css-hlgwow"
              >
                <div
                  class=" css-1dimb5e-singleValue"
                />
                <div
                  class=" css-19bb58m"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="Supplier"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1wy0on6"
              >
                <div
                  aria-hidden="true"
                  class=" css-pogzpp-loadingIndicator"
                >
                  <span
                    class="css-1h1juvu-LoadingDot"
                  />
                  <span
                    class="css-1swgd6a-LoadingDot"
                  />
                  <span
                    class="css-41hts2-LoadingDot"
                  />
                </div>
                <span
                  class=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-1xc3v61-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="Supplier"
              type="hidden"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex gap-4"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="cost"
          >
            Cost
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="cost"
            id="cost"
            name="cost"
            type="number"
            value=""
          />
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="supply_date"
          >
            Supply date
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="supply_date"
            id="supply_date"
            name="supply_date"
            type="date"
            value=""
          />
        </div>
      </div>
      <div
        class="flex gap-4"
      >
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="stock_quantity"
          >
            Stock
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="stock_quantity"
            id="stock_quantity"
            name="stock_quantity"
            type="number"
            value="0"
          />
        </div>
        <div
          class="flex flex-col gap-1 w-full"
        >
          <label
            for="reorder_limit"
          >
            Reorder limit
          </label>
          <input
            class="sc-23ee340d-0 sc-23ee340d-1 cwYaeo jqzeAh"
            data-testid="reorder_limit"
            id="reorder_limit"
            name="reorder_limit"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="flex gap-4"
      >
        <button
          class="sc-7d679c72-0 sc-7d679c72-1 zDViG iTQhfb w-fit"
          type="submit"
        >
          Update
        </button>
        <button
          class="sc-7d679c72-0 sc-7d679c72-1 zDViG iTQhfb w-fit"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Inventory features InventoryPage should match snapshot 1`] = `
<div>
  <div>
    <div>
      <p
        class="text-3xl text-center"
      >
        Inventory
      </p>
      <div
        class="flex gap-8 bg-white w-full p-5 mt-3"
      >
        <input
          class="sc-23ee340d-0 cwYaeo border-0 text-lg"
          placeholder="Search"
          value=""
        />
        <button
          class="sc-7d679c72-0 sc-7d679c72-1 zDViG iTQhfb"
        >
          <p
            class="text-xl"
          >
            +
          </p>
        </button>
      </div>
    </div>
    <table
      class="w-full bg-white mt-3 text-center"
    >
      <thead>
        <tr>
          <th
            class="py-5"
          >
            Product
          </th>
          <th
            class="py-5"
          >
            Supplier
          </th>
          <th
            class="py-5"
          >
            cost
          </th>
          <th
            class="py-5"
          >
            stock
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
